src/Board.cpp

```diff
@@ -0,0 +1,211 @@
+#pragma once
+
+#include "Board.h"
+
+/**
+ * Get the singleton instance of the Board class.
+ *
+ * @return The Board instance.
+ */
+Board& Board::getInstance() {
+    static Board instance;
+    return instance;
+}
+
+/**
+ * Move a piece on the chess board.
+ *
+ * @param moveString The move to be made in string format.
+ * @return An integer representing the result of the move.
+ *         11: The starting position is not occupied by any piece.
+ *         12: The player is trying to move a piece that is not theirs.
+ *         13: The player is trying to capture their own piece.
+ *         21: The move is not valid for the piece.
+ *         31: The move causes the current player to check.
+ *         41: The move results in the current player being in check.
+ *         42: The move is successful and the turn is switched.
+ */
+int Board::move(const std::string& moveString) {
+    auto moveCoords = parseMoveString(moveString);
+    auto& startPos = moveCoords.first;
+    auto& endPos = moveCoords.second;
+    auto& currentPiece = m_chessMatrix[startPos.first][startPos.second];
+
+    if (!currentPiece) {
+        return 11;
```

Comment:
Instead of using magic numbers like 11, 12, etc.,consider defining named constants or enumerations to make the code more readable and maintainable.

Reply:
Done.